### DISCUSSION TITLE
Upgrading bare-metal nodes with rke2 installation

### DISCUSSION THREAD

**Discussion by reisman234** [rancher/rke2#5148] [Q&A] [answered]
Hello

At the moment, I'm making a new installation of the operating system, which runs a bare-metal rke2 cluster  + rook-ceph, and so fare as I currently can tell, It works fine. However, due to the initial setup with a single master, I don't have used a registration address and whatsoever for HA. During the planing for the operating system upgrade, I have now changed worker nodes to master nodes and configured them with the IP of the first initial master.

Now, before I take down the initial master and do some irreversible steps, I want to discuss my next steps. 

At first, I would now change one of the new Masters to be the initial one, by stopping the `rke2-server.service`, deleting the `/etc/rancher/rke2/config.yaml` config-file and turning the service back on. After that, change the server reference of additional master to the new one. After that and from my understanding, it should be safe to take down the old initial master. The image below should make the steps a little more clearer.

![cluster drawio](https://github.com/rancher/rke2/assets/44891585/3e495ea5-c2ff-4432-9321-e44373dfd2b8)

Thanks for any advice!

**Comment by brandond** [comment, marked as answer]
Once servers have joined the cluster, they no longer need to have a server address set. That option is only needed to allow them to find and join the etcd cluster. Once that process is complete, etcd stores the peer member addresses on disk within the datastore, and it automatically connects up to other cluster members on startup, without needing a server address to be set. This means you can actually remove the server address from the server configuration, if you'd like to make the config uniform across your nodes.

**Reply by reisman234** [reply]
Thank you for the response and the clarification. It did work without problems :)
